### ISSUE TITLE
where is the yaml config file?

### ISSUE THREAD

**Issue by runzeer** [microsoft/Oscar#1]

**Comment by xiyinmsu** [comment]
We have saved all running scripts to: https://github.com/microsoft/Oscar/blob/master/MODEL_ZOO.md

Is this what you are looking for?

**Comment by runzeer** [comment]
The run_captioning.py file has the train_yaml config but I have not found this.

**Comment by xiyinmsu** [comment]
the train_yaml is from our dataset, you can download it: https://github.com/microsoft/Oscar/blob/master/DOWNLOAD.md

**Comment by runzeer** [comment]
The file for generating the downloaded files?

**Comment by xiyinmsu** [comment]
To get the image captioning dataset, run this:
wget https://biglmdiag.blob.core.windows.net/oscar/datasets/coco_caption.zip

Then train.yaml is a yaml file specifies the data like features and labels used in training. It comes with the dataset.

**Comment by runzeer** [comment]
Recently, I tried to generate the COCO caption feature files according to your format. The model you used is the bottom-up-attention,right? https://github.com/peteanderson80/bottom-up-attention
If so, there exists some hyperparameters that I can not understand.  I found most of the images in you feature tsv contain 20~30 bounding boxes. So if convenient,could you share the MIN_BOXES , MAX_BOXES as well as the conf_thresh hyperparameter?

**Comment by xjli** [comment]
MIN_BOXES=0, MAX_BOXES=100
conf_thresh = 0.2 or 0.4, I forget the exact value, great probability it is 0.2

**Comment by runzeer** [comment]
The bottom up model is trained in Visual Genome dataset,right?

**Comment by xjli** [comment]
Yes

**Comment by DesaleF** [comment]
I want to test the model with my own images and now I generated the features and labels. however when I run the script oscar/run_captioning.py It needs test.yaml. Do I have to download the whole training data to test the model on my own images? 
This is what I run on my terminal:
python oscar/run_captioning.py \
--do_test --do_eval --test_yaml test.yaml \
--per_gpu_eval_batch_size 64 --num_beams 5 \
--max_gen_length 20 --eval_model_dir oscar/checkpoint-29-66420

**Comment by lisaliu1997** [comment]
> 

I downloaded the dataset but can only see test.label.lineidx, train.feature.lineidx, train.feature.tsv, train.label.lineidx but didn't see any yaml file. Where can I get the yaml file for training? Thanks!

**Comment by DesaleF** [comment]
Yes I have managed to train and evaluate the model on my own custom dataset, I don't have test set in my dataset, I have only train and val set. so the test.yaml file points to the val set.
```
img: val.img.tsv
hw: val.hw.tsv
label: val.label.tsv
feature: val.feature.tsv
caption: val_caption.json
```
this is my test.yaml and val.yaml fil content. 
But first you need to create the .tsv and .json files. If you need help in this regard I can point you the issues which helped me to prepare those files

**Comment by vakadanaveen** [comment]
Can some one tell me how to get only a part of the dataset for training ? 22GB to huge. Also, how to do inference on a downloaded pretrained model ? how to get the test.yaml file ?

**Comment by arushirai1** [comment]
> [DesaleF](/DesaleF)

Hey it would be really nice if you could share those links or an example of what you created. I need to do the same :/

**Comment by arushirai1** [comment]
> Can some one tell me how to get only a part of the dataset for training ? 22GB to huge. Also, how to do inference on a downloaded pretrained model ? how to get the test.yaml file ?

Also just to confirm, do we need the whole download to run inference? I am not sure how the tokenizer works or if its in that massive file.

**Comment by via815via** [comment]
> Yes I have managed to train and evaluate the model on my own custom dataset, I don't have test set in my dataset, I have only train and val set. so the test.yaml file points to the val set.
> 
> ```
> img: val.img.tsv
> hw: val.hw.tsv
> label: val.label.tsv
> feature: val.feature.tsv
> caption: val_caption.json
> ```
> 
> this is my test.yaml and val.yaml fil content. But first you need to create the .tsv and .json files. If you need help in this regard I can point you the issues which helped me to prepare those files

How to get label.tsv? Thanks!
